cpp/src/arrow/array-test.cc

```diff
@@ -58,6 +58,29 @@ TEST_F(TestArray, TestLength) {
   auto data = std::make_shared<PoolBuffer>(pool_);
   std::unique_ptr<Int32Array> arr(new Int32Array(100, data));
   ASSERT_EQ(arr->length(), 100);
+
+  // test slice functions 
+  auto mut_data = data;
+  mut_data->Reserve(100);
+  std::unique_ptr<Int32Array> mut_arr(new Int32Array(25, mut_data));
+
+  auto raw_data = std::dynamic_pointer_cast<MutableBuffer>(mut_arr->data())->mutable_data();
+
+  for(uint32_t i = 0; i < 10; i++) raw_data[i] = i;
```

Comment:
Use `int` here instead of `uint32_t`. https://google.github.io/styleguide/cppguide.html#Integer_Types

Make sure to put braces around the contents of a for loop